// @blueprintjs/datetime/lib/cjs/common/classes.js
"use strict";
/*
 * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
 *
 * Licensed under the terms of the LICENSE file distributed with this project.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DATEINPUT = "pt-dateinput";
exports.DATEINPUT_POPOVER = "pt-dateinput-popover";
exports.DATEPICKER = "pt-datepicker";
exports.DATEPICKER_CAPTION = "pt-datepicker-caption";
exports.DATEPICKER_CAPTION_CARET = "pt-datepicker-caption-caret";
exports.DATEPICKER_CAPTION_MEASURE = "pt-datepicker-caption-measure";
exports.DATEPICKER_CAPTION_SELECT = "pt-datepicker-caption-select";
exports.DATEPICKER_DAY = "DayPicker-Day";
exports.DATEPICKER_DAY_DISABLED = "DayPicker-Day--disabled";
exports.DATEPICKER_DAY_OUTSIDE = "DayPicker-Day--outside";
exports.DATEPICKER_DAY_SELECTED = "DayPicker-Day--selected";
exports.DATEPICKER_FOOTER = "pt-datepicker-footer";
exports.DATEPICKER_MONTH_SELECT = "pt-datepicker-month-select";
exports.DATEPICKER_YEAR_SELECT = "pt-datepicker-year-select";
exports.DATERANGEPICKER = "pt-daterangepicker";
exports.DATERANGEPICKER_CONTIGUOUS = "pt-daterangepicker-contiguous";
exports.DATERANGEPICKER_SINGLE_MONTH = "pt-daterangepicker-single-month";
exports.DATERANGEPICKER_DAY_SELECTED_RANGE = "DayPicker-Day--selected-range";
exports.DATERANGEPICKER_DAY_HOVERED_RANGE = "DayPicker-Day--hovered-range";
exports.DATERANGEPICKER_SHORTCUTS = "pt-daterangepicker-shortcuts";
exports.DATETIMEPICKER = "pt-datetimepicker";
exports.TIMEPICKER = "pt-timepicker";
exports.TIMEPICKER_ARROW_BUTTON = "pt-timepicker-arrow-button";
exports.TIMEPICKER_ARROW_ROW = "pt-timepicker-arrow-row";
exports.TIMEPICKER_DIVIDER_TEXT = "pt-timepicker-divider-text";
exports.TIMEPICKER_HOUR = "pt-timepicker-hour";
exports.TIMEPICKER_INPUT = "pt-timepicker-input";
exports.TIMEPICKER_INPUT_ROW = "pt-timepicker-input-row";
exports.TIMEPICKER_MILLISECOND = "pt-timepicker-millisecond";
exports.TIMEPICKER_MINUTE = "pt-timepicker-minute";
exports.TIMEPICKER_SECOND = "pt-timepicker-second";
exports.TIMEPICKER_AMPM_SELECT = "pt-timepicker-ampm-select";
